cpp/tests/reshape/table_to_array.cpp

```diff
@@ -0,0 +1,148 @@
+#include <cudf_test/column_wrapper.hpp>
+#include <cudf_test/type_list_utilities.hpp>
+#include <cudf_test/type_lists.hpp>
+
+#include <cudf/column/column_factories.hpp>
+#include <cudf/reshape.hpp>
+#include <cudf/table/table.hpp>
+#include <cudf/table/table_view.hpp>
+#include <cudf/utilities/default_stream.hpp>
+#include <cudf/utilities/type_dispatcher.hpp>
+
+#include <rmm/cuda_stream_view.hpp>
+#include <rmm/device_buffer.hpp>
+
+template <typename T>
+struct TableToDeviceArrayTypedTest : public cudf::test::BaseFixture {};
+
+using SupportedTypes = cudf::test::Types<int8_t,
+                                         int16_t,
+                                         int32_t,
+                                         int64_t,
+                                         uint8_t,
+                                         uint16_t,
+                                         uint32_t,
+                                         uint64_t,
+                                         float,
+                                         double,
+                                         cudf::timestamp_D,
+                                         cudf::timestamp_s,
+                                         cudf::timestamp_ms,
+                                         cudf::timestamp_us,
+                                         cudf::timestamp_ns,
+                                         cudf::duration_D,
+                                         cudf::duration_s,
+                                         cudf::duration_ms,
+                                         cudf::duration_us,
+                                         cudf::duration_ns>;
+
+TYPED_TEST_SUITE(TableToDeviceArrayTypedTest, SupportedTypes);
+
+TYPED_TEST(TableToDeviceArrayTypedTest, SupportedTypes)
+{
+  using T     = TypeParam;
+  auto stream = cudf::get_default_stream();
+
+  auto const dtype = cudf::data_type{cudf::type_to_id<T>()};
+
+  auto const col0 = cudf::test::make_type_param_vector<T>({1, 2, 3});
+  auto const col1 = cudf::test::make_type_param_vector<T>({4, 5, 6});
+  auto const col2 = cudf::test::make_type_param_vector<T>({7, 8, 9});
+  auto const col3 = cudf::test::make_type_param_vector<T>({10, 11, 12});
+
+  std::vector<std::unique_ptr<cudf::column>> cols;
+  auto make_col = [&](auto const& data) {
+    return std::make_unique<cudf::column>(
+      cudf::test::fixed_width_column_wrapper<T>(data.begin(), data.end()));
+  };
+
+  cols.push_back(make_col(col0));
+  cols.push_back(make_col(col1));
+  cols.push_back(make_col(col2));
+  cols.push_back(make_col(col3));
+
+  cudf::table_view input({cols[0]->view(), cols[1]->view(), cols[2]->view(), cols[3]->view()});
+  size_t num_elements = 3 * 4;
+  rmm::device_buffer output(num_elements * sizeof(T), stream);
+
+  cudf::table_to_device_array(
+    input, output.data(), dtype, stream, rmm::mr::get_current_device_resource());
+
+  std::vector<T> host_result(num_elements);
+  CUDF_CUDA_TRY(cudaMemcpy(
```

Comment:
I think we can use a vector factory (https://github.com/rapidsai/cudf/blob/branch-25.06/cpp/include/cudf/detail/utilities/vector_factories.hpp) to simplify this code. We try to avoid raw `cudaMemcpy` calls!

Reply:
I used `make_zeroed_device_uvector_sync`